### DISCUSSION TITLE
prepaid, long-expiry hotspot

### DISCUSSION THREAD

**Discussion by sgbaird** [sparks-baird/self-driving-lab-demo#83] [Demo extensions and design]
UPDATE: I recommend using a [Hologram SIM card](https://store.hologram.io/store/) with [a Cudy 2023 AC1200 Dual Band 4G LTE Modem Router](https://www.amazon.com/Cudy-Unlocked-1200Mbps-EC25-AFX-Cloudflare/dp/B089JW8H4N); however, this can get quite expensive when passing large amounts of data back-and-forth. This has been working well for me with minimal issues for several months. If you are located outside of the US, you can check to see if the Cudy router is available in your country's Amazon by replacing `.com` with the appropriate extension (Canada: `.ca` | Germany: `.de` | UK: `.co.uk`) in the URL below.

> https://www.amazon.com/dp/B089JW8H4N

Alternatively, you can source a SIM-enabled Cudy router from a different supplier, choose a different SIM-enabled WiFi router, or work with your local wireless providers (e.g., Verizon, AT&T, T-Mobile, Bell, Rogers) to explore IoT-style options. Often, these require you purchasing a somewhat expensive WiFi device and/or signing a multi-year contract.

If you are just trying to get something up-and-running temporarily, you can use a phone hotspot in extended compatibility mode (and with "limit IP address tracking" turned off if applicable). See https://github.com/sparks-baird/self-driving-lab-demo/issues/76.

EDIT: The following might help out with considering options for a 2.4 GHz WPA2-personal WiFi network:
<img src="https://github.com/user-attachments/assets/633a740d-b7df-4baa-9d7e-ce3ec729258f" width=700>

**Comment by sgbaird** [comment]
https://www.reddit.com/r/NoContract/comments/bzsm2w/hotspot_data_with_no_required_monthly_fee_that/

Is there something cheaper than TravelWiFi (~150 USD, 1 GB preloaded, not sure about expiration)? https://travelwifi.com/portable-hotspots/sapphire-2

Maybe the Helium network could be of use here https://docs.helium.com/use-the-network/

**Comment by sgbaird** [comment]
@calcut, you mentioned https://blues.io in https://github.com/adafruit/circuitpython/issues/6988#issuecomment-1266695981. Does this seem like something that would work well as a low-cost, prepaid WiFi hotspot?

related https://github.com/sparks-baird/self-driving-lab-demo/issues/76

**Reply by calcut** [reply]
Hmm, not sure. If the pre-paid, pre-activated bit is what you care about, then maybe!
You'd have to link up a cellular notecard with something like an ESP32 feather with wifi in access point mode.
if your data usege is gonna be low enough, could work. Blues only comes with 500MB to last 10 years.

For a plain 4G/Wifi hotspot Ive used these before https://www.amazon.co.uk/dp/B016ZWXYXG/
But its not battery powered or pre-paid.

**Comment by sgbaird** [comment]
A response from travelwifi about 2.4 GHz connectivity for [Sapphire 2](https://travelwifi.com/portable-hotspots/sapphire-2/):

> The Sapphire 3 supports 2.4 GHz network and it does not require any special setup for it to function. Please inquire from the providers of the Pico W microcontroller if is compatible with the mifi hotspots.

**Comment by sgbaird** [comment]
## Summary of Options

Choose one mobile hotspot or one (SIM Card + Router).

| Device | Cost | Data | Number of Connections* | Notes |
| ---- | ---- | ---- | ---- | ---- |
| [KeepGo Mobile Hotspot](https://www.keepgo.com/products/keepgo-lifetime-mobile-travel-wifi-hotspot) | $129 | $3/100 MB | 15 | needs to be refilled with at least $3 once per year. Lack of customizability (e.g., due to use of captive portal) mentioned in [an Amazon review](https://www.amazon.com/gp/customer-reviews/RYR0EHG88XPKV/ref=cm_cr_dp_d_rvw_ttl?ie=UTF8&ASIN=B084BQK3HP). |
| [Hologram Global IoT SIM Card](https://store.hologram.io/store/) | $5.00 | $8/100 MB | N/A | $0.70/month/device = $8.40/year ([source](https://www.hologram.io/pricing/)) |
| [KeepGo SIM Card](https://www.keepgo.com/products/keepgo-lifetime-prepaid-data-sim-card) | $49 | $3/100 MB | N/A | needs to be refilled with at least $3 once per year. **See notes about hotspot vs. SIM card |
| [Things Mobile IoT SIM Card](https://www.thingsmobile.com/business/plans/pay-per-use-plan-without-subscription-fee) | $3.00 | $10/100 MB | N/A | Hidden monthly fees when reading fine print: $3.75/month |
| [Cudy N300 WiFi Unlocked 4G LTE Modem Router with SIM Card Slot](https://www.amazon.com/Cudy-Unlocked-EC25-AFX-Qualcomm-Cloudflare/dp/B085S2147W/) | $89.90 | N/A | [30+](https://www.syntech.co.za/product/cudy-n300-lte-4g-outdoor/) | at least 6 device connections based on one of the reviewer's experiences, but probably 30+ based on [syntech datasheet](https://www.syntech.co.za/product/cudy-n300-lte-4g-outdoor/) |
| [Router Alcatel Link Hub 4G LTE Unlocked](https://www.amazon.com/Alcatel-Unlocked-Worldwide-Caribbean-HH41NH-2BTGMXA-1/dp/B07RNDJFXN) | $82.99 | N/A | 32 | Some reviews mention worsening device connectivity issues (months?) |
| [Alcatel Link Zone 4G LTE Mobile Router](https://www.amazon.com/Alcatel-MW41NF-2AOFUS1-Hotspot-Unlocked-Caribbean/dp/B084BQK3HP) (SIM not included) | $59.99 | N/A | 15 | This is what is used for KeepGo Mobile Hotspot |
| [TravelWiFi Sapphire2 Mobile Hotspot](https://travelwifi.com/portable-hotspots/sapphire-2/) | $129 | 1 GB preloaded | 5 | Not clear, but seems like there aren't pay-as-you-go data plans, so I wonder if the 1 GB expires after 30 days |

Might be worth getting the Sapphire2 and trying to connect many devices to it. EDIT: based on some dialogue with travelwifi support, it seems like the maximum number of devices is imposed, rather than just being a recommendation.
> Yes we do try that basically your device can't connect to 6th device as your device can only connect to 5 device Maximum at a time.

*Number of connections might be a non-issue, as it seems that [the physical limit of device connections is around 250](https://www.quora.com/How-many-devices-can-2-4-GHz-support-and-what-happens-if-the-devices-are-more-than-that). The issue is more with splitting the bandwidth across too many devices, which isn't much of a problem for such small data transfer for these demos.

**KeepGo mobile hotspot vs. SIM card + router (below talks about a mobile device)
> ** Tethering works in most countries with most mobile device models. Because it depends on the local provider and the specific device model we cannot guarantee it will work everywhere. On the other hand our mobile wifi hotspot does work everywhere. If you are planning to travel with several WiFi-enabled devices, you can opt for the [Keepgo Mobile WiFi Hotspot](https://www.keepgo.com/products/wifi_hotspot_1gb_europe_asia_america) to guarantee that you will be able to connect all your devices (up to 16) to the internet in any location. Keep in mind that Mobile WiFi Hotspot has a few advantages over tethering on your phone:
> 1. You spare your phone’s battery;
> 1. You can reliably tether multiple devices;
> 1. You can work for longer periods without drops or throttling.
> 1. You can keep your original SIM in your phone

No-go options that I looked at:
- EIOTClUB SIM, too restrictive, no router-based prepaid plan

**Comment by sgbaird** [comment]
A Hologram SIM card with [a Cudy 2023 AC1200 Dual Band 4G LTE Modem Router](https://www.amazon.com/Cudy-Unlocked-1200Mbps-EC25-AFX-Cloudflare/dp/B089JW8H4N) has been working fairly well. The internet dropped off randomly after a month or so. Turning the device off and then on again didn't seem to help. The Hologram SIM was working fine according to the online portal. It wasn't clear why the router didn't have internet, and I ended up having to go back onto https://cudy.net while connected to the network and "save and apply" the cellular network settings, even though I didn't change anything about the settings. It's back up-and-running again.

**Reply by sgbaird** [reply]
This has still been working well. A few additional notes:

While I can set a data limit on hologram, and the interface is slick, it's easy to go over the data limits. Likewise, an existing "session" can continue to use data past the limit (i.e., it's not an immediate shut-off). However, on the Cudy router, there is more robust control. Data limits can be set, per-device settings can be set, etc. Unfortunately, the bandwidth couldn't be set to less than 1 mpbs.

Strangely, the Kasa KP400 seemed to be eating up a ton of data.

Oftentimes, I've needed to set up equipment by connecting to this network via phone or laptop. If on laptop, it's best if you can have an Ethernet network connected to your computer, which seems to do a good job of taking the bulk of traffic. This way, you can still have web surfing level of internet access while communicating with/setting up equipment.

For mobile, there are two things you should do:
1. Configure the network to not "auto connect"
2. Set the network as a metered network, which prevents the device from consuming a lot of background data
3. Turn data saver on (optional)

This works on Android for a Pixel 4 for me.

It may be worth having an unlimited data plan SIM card for prototyping or when network traffic is high, but reliability isn't as much of a concern.

**Reply by sgbaird** [reply]
Hologram isn't a great option for me to use with camera-enabled printers like the FormAuto package or BambuLabs, since the data usage is really high. This has been the case for others as well. Volume discount pricing only really starts to kick in at around US$500/month (not automatically, but by working directly with hologram to get a plan).

Likewise, putting a standard cell phone SIM card into a router may violate the terms of service from the network provider.
